src/v/cloud_storage/remote_path_provider.cc

```diff
@@ -32,30 +32,37 @@ remote_path_provider remote_path_provider::copy() const {
 
 ss::sstring remote_path_provider::topic_manifest_prefix(
   const model::topic_namespace& topic) const {
+    const auto& tp_ns = _topic_namespace_override.value_or(topic);
```

Comment:
nit: please squash this with the commit that adds tests for the remote_path_provider

Reply:
Squashed commits.